### ISSUE TITLE
Numpy array real and imag don't seem to be visible to Pyright

### ISSUE THREAD

**Issue by NeilGirdhar** [microsoft/pyright#3385]
### Describe the issue:

Array `real` and `imag` don't seem to be visible in Pyright

### Reproduce the code example:
```python
import numpy as np
x = np.zeros(10, dtype=np.float64)
print(x.real)
```

### Error message:
In Pyright, this gives:
```shell
error: Cannot access member "real" for type "NDArray[float64]"
```

### NumPy/Python version information:
1.22.3 3.10.4 (main, Apr  7 2022, 10:39:08) [GCC 11.2.0]

See https://github.com/numpy/numpy/issues/21393

**Comment by JelleZijlstra** [comment]
Likely related to the unusual self types here: https://github.com/numpy/numpy/blob/fd646bd693400ce408566674f266407672248cad/numpy/__init__.pyi#L1484

**Comment by erictraut** [comment]
I've narrowed it down to this self-contained sample:

```python
from typing import Protocol, TypeVar

_T_co = TypeVar("_T_co", covariant=True)
_Self = TypeVar("_Self")

class number:
    @property
    def real(self: _Self) -> _Self: ...

class _SupportsReal(Protocol[_T_co]):
    @property
    def real(self) -> _T_co: ...

num = number()
x: _SupportsReal[number] = num
```

There is no error reported if the `@property` is removed, so this appears to be a problem related to the special-case logic for properties.

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.243, which I just published. It will also be included in the next release of pylance.
